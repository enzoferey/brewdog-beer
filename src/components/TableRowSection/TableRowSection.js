import React from "react";
import PropTypes from "prop-types";
import style from "./table-row-section.scss";

const TableRowSection = ({ children }) => (
  <div className={style.main}>{children}</div>
);

TableRowSection.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
};

export default TableRowSection;
