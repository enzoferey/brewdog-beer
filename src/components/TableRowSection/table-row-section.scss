.main {
  flex: 1;
  border-right: 1px solid #707070;
  display: flex;
  justify-content: center;
  align-items: center;
  margin: 8px 0;
  padding: 0 13px;
}
