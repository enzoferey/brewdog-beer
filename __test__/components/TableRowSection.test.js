import React from "react";
import TableRowSection from "components/TableRowSection";

const props = {
  flex: 2,
};

const MyText = () => <p>hi</p>;

test("renders correctly", () => {
  testSnapshot(<TableRowSection />);
});

test("applies `flex` prop correctly", () => {
  testSnapshot(<TableRowSection {...props} />);
});

test("behaves like a HOC", () => {
  const wrapper = shallow(
    <TableRowSection>
      <MyText />
    </TableRowSection>
  );
  expect(wrapper.find(MyText).length).toBe(1);
});
