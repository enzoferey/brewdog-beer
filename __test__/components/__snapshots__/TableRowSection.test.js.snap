// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies \`flex\` prop correctly 1`] = `
<div
  className={undefined}
  style={
    Object {
      "flex": 2,
    }
  }
/>
`;

exports[`renders correctly 1`] = `
<div
  className={undefined}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
